odoo/addons/base/views/res_currency_views.xml

```diff
@@ -66,7 +66,13 @@
             <field name="model">res.currency</field>
             <field name="arch" type="xml">
                 <search string="Currencies">
-                    <field name="name" string="Currency"/>
+                    <field name="name" string="Currency" filter_domain="('|','|','|','|',
```

Comment:
Why not implementing `_name_search` instead?

Reply:
Have a look at https://github.com/odoo/odoo/commit/3155c3e425581b71491844e7f9a3dd76a9f245a4